pkg/iac/scanners/terraform/parser/modules_test.go

```diff
@@ -0,0 +1,71 @@
+package parser
+
+import (
+	"context"
+	"path"
+	"testing"
+
+	"github.com/aquasecurity/trivy/internal/testutil"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+)
+
+func TestFindRootModules(t *testing.T) {
+	tests := []struct {
+		name     string
+		files    map[string]string
+		expected []string
+	}{
+		{
+			name: "multiple root modules",
+			files: map[string]string{
+				"code/main.tf": `
+module "this" {
+  count = 0
+  source = "./modules/s3"
+}`,
+				"code/modules/s3/main.tf": `
+module "this" {
+  source = "./modules/logging"
```

Comment:
Does the parsing logic still work if this was:

```suggestion
  source = "modules/logging"
```

It is possible to define file paths without mentioning the present dir.

Reply:
>A local path must begin with either ./ or ../ to indicate that a local path is intended, to distinguish from [a module registry address](https://developer.hashicorp.com/terraform/language/modules/sources#terraform-registry).

https://developer.hashicorp.com/terraform/language/modules/sources#local-paths

Reply:
I see, is it possible to have a root module that's remote/registry address?

Reply:
@simar7 No, the root can only be local